src/harness/parallel/worker.ts

```diff
@@ -2,7 +2,9 @@ namespace Harness.Parallel.Worker {
     let errors: ErrorInfo[] = [];
     let passing = 0;
 
-    type Executor = {name: string, callback: Function, kind: "suite" | "test"} | never;
+    type MochaCallback = (this: Mocha.ITestCallbackContext, done: MochaDone) => void;

```

Comment:
Technically, `this` should be an ISuiteCallbackContext for "suite" and only this context for "test". Also gives you your answer for the todo below: this callback type (as written, not corrected) would be correct there, I believe.